Tools/bat/chlauthmenu.bat

```diff
@@ -8,41 +8,68 @@
 :: 
 :: 
 :: ============================================================================
-:: 20161017 T.Rob    New script
-:: 
-:: 
+:: 20161017 T.Rob    	New script
+:: 20161026 fjbsaper	made elevated privilege independant of MQ
+::                      changed testuser to %testuser%
+:: 20161026 T.Rob       Changed to use logged-in user as the MQCD User ID
+::                      Tested that USERNAME is set and != testuser
+::                      Restored USERSRC(NOACCESS) rules for logged-in user
+::                      Moved listener definition to the initialization subroutine
+::                      Added comments for each subroutine
+::                      
 :: ============================================================================
 
 
-:: Arrange to discard command line history from the menu keystrokes
 @echo off
-if "%~1"==":historySafe" goto :historySafe
-cmd /d /c "%~f0" :historySafe
+:: Make sure user name environment variable is set
+:: Is this ever *not* set in Windows?
+if "a%USERNAME%a" == "aa" (
+	echo Environment variable USERNAME not set!
+	goto usage
+)
+
+:: check parameters
+set testuser=%1
+if "a%testuser%a" == "aa" (
+	goto usage
+)
+if "a%testuser%a" == "aa%USERNAME%a" (
+	goto usage
+)
+
+
+:: Arrange to discard command line history from the menu keystrokes
+if "%~2"==":historySafe" goto :historySafe
+cmd /d /c "%~f0" %1 :historySafe
 exit /b
 
 :historySafe
+echo testuser=%testuser%
 
 :: Main program begins
 SETLOCAL ENABLEEXTENSIONS
 SETLOCAL ENABLEDELAYEDEXPANSION
 SETLOCAL
 
-:: Test to see if user has WMQ admin privileges
-set Response=
-set isAdmin=
-for /f "tokens=*" %%a in ('dspmqtrn 2^>^&1') do @set Response=%%a
 
-set isAdmin=!Response:~0,7!
+:: Test to see if user has elevated WMQ admin privileges
+:: we are running strmqm which does require elevated privileges in windows
+:: set Response=
+:: for /f "tokens=*" %%a in ('dspmqtrn 2^>^&1') do @set Response=%%a
+:: set isAdmin=!Response:~0,7!
+:: if NOT "%isAdmin%"=="AMQ7028" (
 
-if NOT "%isAdmin%"=="AMQ7028" (
+net session > nul 2>&1
+set /a isAdmin=%errorlevel%
+if %isAdmin% gtr 0 (
 	echo. 
-	echo %0: This script must be run from an account with WeSphere MQ adminstrator privileges.
+	echo %0: This script must be run from an account with elevated adminstrator privileges ^(run as administrator^).
 	exit /B 2
 )
```

Comment:
You might want to revert this part.
On Pro it will work fine. On Windows home it will be the equivalent of running with elevated privileges... and this will catch it.